Content.Client/UserInterface/CharacterSetupGui.cs

```diff
@@ -0,0 +1,267 @@
+                Text = "Create new slot...",
+                ToolTip = $"A maximum of {preferencesManager.Settings.MaxCharacterSlots} characters are allowed."
+            };
+            _createNewCharacterButton.OnPressed += args =>
+            {
+                preferencesManager.CreateCharacter(HumanoidCharacterProfile.Default());
+                UpdateUI();
+            };
+
+            hBox.AddChild(new PanelContainer
+            {
+                PanelOverride = new StyleBoxFlat {BackgroundColor = NanoStyle.NanoGold},
+                CustomMinimumSize = (2, 0)
+            });
+            _humanoidProfileEditorPanel = new HumanoidProfileEditorPanel(localization,
+                resourceCache,
+                preferencesManager
+            );
+            _humanoidProfileEditorPanel.OnProfileChanged += newProfile => { UpdateUI(); };
+            hBox.AddChild(_humanoidProfileEditorPanel);
+
+            UpdateUI();
+        }
+
+        private void UpdateUI()
+        {
+            var numberOfFullSlots = 0;
+            var characterButtonsGroup = new ButtonGroup();
+            _charactersVBox.RemoveAllChildren();
+            var characterIndex = 0;
+            foreach (var character in _preferencesManager.Preferences.Characters)
+            {
+                if (character is null)
+                {
+                    characterIndex++;
+                    continue;
+                }
+
+                numberOfFullSlots++;
+                var characterPickerButton = new CharacterPickerButton(_entityManager,
+                    _preferencesManager,
+                    characterButtonsGroup,
+                    character,
+                    character.Name,
+                    character.Name,
+                    "Assistant");
+                _charactersVBox.AddChild(characterPickerButton);
+
+                var characterIndexCopy = characterIndex;
+                characterPickerButton.ActualButton.OnPressed += args =>
+                {
+                    _humanoidProfileEditorPanel.Profile = (HumanoidCharacterProfile) character;
+                    _humanoidProfileEditorPanel.CharacterSlot = characterIndexCopy;
+                    _humanoidProfileEditorPanel.UpdateControls();
+                    _preferencesManager.SelectCharacter(character);
+                };
+                characterIndex++;
+            }
+
+            _createNewCharacterButton.Disabled =
+                numberOfFullSlots >= _preferencesManager.Settings.MaxCharacterSlots;
+            _charactersVBox.AddChild(_createNewCharacterButton);
+        }
+
+        private class CharacterPickerButton : Control
+        {
+            public readonly Button ActualButton;
+
+            public CharacterPickerButton(
+                IEntityManager entityManager,
+                IClientPreferencesManager preferencesManager,
+                ButtonGroup group,
+                ICharacterProfile profile,
+                string slotName,
```

Comment:
I need to change this method to read these 3 values from `profile` instead, refactoring mishap